telegram_data_downloader/processor/message_downloader.py

```diff
@@ -0,0 +1,233 @@
+        fwd_from = (
+            telethon.utils.get_peer_id(message.fwd_from.from_id)
+            if message.fwd_from and message.fwd_from.from_id
+            else None
+        )
+        from_id = (
+            telethon.utils.get_peer_id(message.from_id) if message.from_id else None
+        )
+
+        msg_attributes: MessageAttributes = {
+            "id": message.id,
+            "date": message.date,
+            "from_id": PeerID(from_id) if from_id else None,
+            "fwd_from": PeerID(fwd_from) if fwd_from else None,
+            "message": message.message or "",
+            "type": MessageType.TEXT,
+            "duration": None,
+            "to_id": PeerID(telethon.utils.get_peer_id(message.to_id)),
+            "reactions": {},
+        }
+
+        if media := message.media:
+            # For stickers, videos, and voice messages
+            if (
+                isinstance(media, tl_types.MessageMediaDocument)
+                and media.document
+                and not isinstance(media.document, tl_types.DocumentEmpty)
+            ):
+                for attribute in media.document.attributes:
+                    if isinstance(attribute, tl_types.DocumentAttributeSticker):
+                        msg_attributes["message"] = attribute.alt
+                        msg_attributes["type"] = MessageType.STICKER
+                        break
+
+                    elif isinstance(attribute, tl_types.DocumentAttributeVideo):
+                        msg_attributes["duration"] = attribute.duration
+                        msg_attributes["type"] = MessageType.VIDEO
+                        break
+
+                    elif (
+                        isinstance(attribute, tl_types.DocumentAttributeAudio)
+                        and attribute.voice
+                    ):
+                        msg_attributes["duration"] = attribute.duration
+                        msg_attributes["type"] = MessageType.VOICE
+                        break
+            elif isinstance(message.media, tl_types.MessageMediaPhoto):
+                msg_attributes["type"] = MessageType.PHOTO
+
+        return msg_attributes
+
+    @async_retry(
+        telethon.errors.common.InvalidBufferError,
+        base_sleep_time=5.0,
+        max_tries=5,
+    )
+    async def _get_message_reactions(
+        self, message: TLMessage, dialog_peer: tl_types.TypeInputPeer
+    ) -> dict[PeerID, tl_types.ReactionEmoji]:
+        try:
+            result: tl_types.messages.MessageReactionsList = await self.client(
+                telethon.functions.messages.GetMessageReactionsListRequest(
+                    peer=dialog_peer,
+                    id=message.id,
+                    limit=self.reactions_limit_per_message,
+                )
+            )  # type: ignore
+        except telethon.errors.BroadcastForbiddenError:
+            # logger.debug("channel is broadcast: cannot retrieve reactions from message")
```

Comment:
why did you comment this?

Reply:
This log message was added before the check on broadcast channel was used, so it clogged the entire log file, but not it is indeed redundant

Reply:
But the log message with "message not found" is actually useful in case of deleted messages or some error on Telegram side, so I'd keep it